src/highlighter.js

```diff
@@ -102,6 +104,21 @@ export default function (options) {
         return $(containerSelector).find(`.${highlightingClasses.join(',.')}`);
     }
 
+    /**
+     * Attach data to wrapper node.
+     * Use it when deleting this highlight to know if highlight content should be merged with neighbour text nodes or not.
+     * Needed to keep markup the same as it was before highlighting.
+     * @param {HTMLElement} node
+     * @param {Boolean} beforeWasSplit
+     * @param {Boolean} afterWasSplit
+     */
+    function addSplitData(node, beforeWasSplit, afterWasSplit) {
+        if (keepEmptyNodes) {
```

Comment:
I'm not sure about putting that condition _inside_ this function. Since you are checking `if (keepEmptyNodes)` (or its inverse) some 10 times in the module, may as well add it _around_ `addSplitData()` in the 2-3 places where needed. The number of conditions in the code would increase, but it seems easier to understand when they are all on the same level. WDYT?

Reply:
Agree